Index: codereview/library.py
===================================================================
--- codereview/library.py	(Revision 315)
+++ codereview/library.py	(Arbeitskopie)
@@ -28,7 +28,7 @@
 register = django.template.Library()
 
 
-@register.filter
+#@register.filter
 def nickname(email, arg=None):
   """Render an email address or a User object as a nickname.
 
@@ -58,7 +58,7 @@
   return ', '.join(nickname(email, arg) for email in email_list)
 
 
-@register.filter
+#@register.filter
 def show_user(email, arg=None, autoescape=None, memcache_results=None):
   """Render a link to the user's dashboard, with text being the nickname."""
   if isinstance(email, users.User):
