Index: codereview/models.py
===================================================================
--- codereview/models.py	(revision 427)
+++ codereview/models.py	(working copy)
@@ -536,7 +536,8 @@
     if account is not None:
       return account
     nickname = cls.create_nickname_for_user(user)
-    return cls.get_or_insert(key, user=user, email=email, nickname=nickname,
+    return cls.get_or_insert(key, user=user, email=str(email),
+                             nickname=str(nickname),
                              fresh=True)
 
   @classmethod
