Index: codereview/views.py
===================================================================
--- codereview/views.py	(revision 439)
+++ codereview/views.py	(working copy)
@@ -655,6 +655,8 @@
   issues = query.fetch(limit+1, offset)
   more = bool(issues[limit:])
   if more:
+    # item deletion not supported by Django's QuerySet.
+    issues = list(issues)
     del issues[limit:]
   if more:
     next = '/all?offset=%d&limit=%d' % (offset+limit, limit)
@@ -2175,7 +2177,10 @@
         # Get the patch key value without loading the patch entity.
         # NOTE: Unlike the old version of this code, this is the
         # recommended and documented way to do this!
-        pkey = models.Comment.patch.get_value_for_datastore(c)
+        # pkey = models.Comment.patch.get_value_for_datastore(c)
+        # FIXME(andi): see Issue 1
+        #   http://code.google.com/p/django-gae2django/issues/detail?id=1
+        pkey = getattr(c, '_patch', None)
         if pkey in patches:
           patch = patches[pkey]
           c.patch = patch
